backend/python/app/services/implementations/entity_service_mg.py

```diff
@@ -0,0 +1,56 @@
+from ...models.entity_mg import Entity
+from ..interfaces.entity_service import IEntityService
+
+
+class EntityService(IEntityService):
+    def __init__(self, logger):
+        self.logger = logger
+
+    def get_entities(self):
+        entities = []
+
+        for result in Entity.objects:
+            result_dict = result.to_serializable_dict()
+            entities.append(result_dict)
+
+        return entities
+
+    def get_entity(self, id):
+        entity = Entity.objects.get(id=id)
+
+        if entity is None:
+            self.logger.error("Invalid id")
+            raise Exception("Invalid id")
+
+        return entity.to_serializable_dict()
+
+    def create_entity(self, entity):
+        try:
+            new_entity = Entity(**entity)
+            new_entity.save()
+        except Exception as error:
+            self.logger.error(str(error))
+            raise error
+
+        return new_entity.to_serializable_dict()
+
+    def update_entity(self, id, entity):
+        updated_entity = Entity.objects.get(id=id)
```

Comment:
```suggestion
        updated_entity = Entity.objects(id=id).modify(new=True, **entity)
```

This will atomically find and update, https://docs.mongoengine.org/guide/querying.html#atomic-updates